staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go

```diff
@@ -205,9 +204,19 @@ func (s *WatchServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 			unknown.Raw = buf.Bytes()
 			event.Object = &unknown
 
-			// the internal event will be versioned by the encoder
-			*internalEvent = metav1.InternalEvent(event)
-			if err := e.Encode(internalEvent); err != nil {
+			// create the external type directly and encode it.  Clients will only recognize the serialization we provide
+			// so there isn't any benefit to doing separate conversion and encoding.
+			externalEvent := &metav1.WatchEvent{}
+			externalEvent.Type = string(event.Type)
```

Comment:
I don't feel that strongly, would just be nice to have something that will smack us here if this breaks.